### PR TITLE
[luafilesystem][lpeg] Bump versions

### PR DESCRIPTION
**Pull request by TheCycoONE**
**Describe the pull request**
Bump versions of luafilesystem and lpeg to the latest available

### CODE REVIEW

File: ports/lpeg/portfile.cmake [hunk 1]
```diff
@@ -1,12 +1,12 @@
 #cmake-only scripts
 include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/lpeg/portfile.cmake:2]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/luafilesystem/portfile.cmake [hunk 2]
```diff
@@ -1,17 +1,15 @@
 include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/luafilesystem/portfile.cmake:1]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/luafilesystem/portfile.cmake [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
     OUT_SOURCE_PATH SOURCE_PATH
     REPO keplerproject/luafilesystem
     REF ${LUAFILESYSTEM_REVISION}
     SHA512 ${LUAFILESYSTEM_HASH}
     HEAD_REF master
-    PATCHES
-        lfs-def-fix.patch
 )
 
 file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})
```
Comment (LilyWangL) [inline ports/luafilesystem/portfile.cmake:15]: Can you please use command `file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by TheCycoONE** [comment]
Pushed the requested changes. Thanks @LilyWangL

**Comment by TheCycoONE** [comment]
I've removed the windows only gates. That was left over from when vcpkg only supported windows, nothing about these packages should be Windows only.

**Review by LilyWangL — APPROVED** [review]

**Pull request merged**